src/lib/sw_engine/tvgSwRenderer.cpp

```diff
@@ -92,46 +88,43 @@ struct SwShapeTask : SwTask
             uint8_t alpha = 0;
             sdata->fillColor(nullptr, nullptr, nullptr, &alpha);
             alpha = static_cast<uint8_t>(static_cast<uint32_t>(alpha) * opacity / 255);
-            bool renderShape = (alpha > 0 || sdata->fill());
-            if (renderShape || validStroke) {
+            visibleFill = (alpha > 0 || sdata->fill());
+            if (visibleFill || visibleStroke) {
                 shapeReset(&shape);
                 if (!shapePrepare(&shape, sdata, transform, clipRegion, bbox, mpool, tid)) goto err;
-                if (renderShape) {
-                    /* We assume that if stroke width is bigger than 2,
-                       shape outline below stroke could be full covered by stroke drawing.
-                       Thus it turns off antialising in that condition.
-                       Also, it shouldn't be dash style. */
-                    auto antiAlias = (strokeAlpha == 255 && strokeWidth > 2 && sdata->strokeDash(nullptr) == 0) ? false : true;
-                    if (!shapeGenRle(&shape, sdata, antiAlias, clips.count > 0 ? true : false)) goto err;
```

Comment:
This looks wrong, shape rle is necessary even though only the stroking is valid but you changed it to when the filling is valid....